app/server/src/main/scala/org/bitcoins/server/Main.scala

```diff
@@ -219,9 +220,35 @@ object Main extends App with HttpLogger {
           tmpWallet)
         _ = logger.info("Starting wallet")
 
-        zmq = BitcoindRpcBackendUtil.createZMQWalletCallbacks(wallet)
+        blockCount <- bitcoind.getBlockCount
+        // Create callbacks for processing new blocks
+        _ = bitcoindRpcConf.zmqPortOpt match {
+          case Some(_) =>
+            val zmq = BitcoindRpcBackendUtil.createZMQWalletCallbacks(wallet)
+            zmq.start()
+          case None =>
+            var prevCount = blockCount
```

Comment:
This should really be moved out of `Main.scala` and it should be unit tested in the wallet project?